lib/ui/key.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.12
+
+part of dart.ui;
+
+/// How the key has changed since the last report.
+///
+// Must match the KeyChange enum in ui/window/key_data.h.
+enum KeyChange {
+  /// The key is pressed.
+  down,
+
+  /// The key is released.
+  up,
+
+  /// The key is held, causing a repeated key input.
+  repeat,
+}
+
+/// Information about the change of a key.
+class KeyData {
+  /// Creates an object that represents the change of a key.
+  const KeyData({
+    required this.timeStamp,
+    required this.change,
+    required this.physical,
+    required this.logical,
+    required this.character,
+    required this.synthesized,
+  });
+
+  /// Time of event dispatch, relative to an arbitrary timeline.
+  ///
+  /// For [KeyChange.synchronize] and [KeyChange.cancel] events, the [timeStamp]
```

Comment:
Weren't `KeyChange.synchronized` and `KeyChange.cancel` replaced by the `synthesized` property on this class?